src/mapper/java/org/codehaus/jackson/map/deser/SecurityBeanDeserializer.java

```diff
@@ -0,0 +1,73 @@
+package org.codehaus.jackson.map.deser;
+
+
+import org.codehaus.jackson.map.BeanDescription;
+import org.codehaus.jackson.map.BeanProperty;
+import org.codehaus.jackson.map.DeserializationConfig;
+import org.codehaus.jackson.map.DeserializerProvider;
+import org.codehaus.jackson.map.Deserializers;
+import org.codehaus.jackson.map.JsonDeserializer;
+import org.codehaus.jackson.map.JsonMappingException;
+import org.codehaus.jackson.type.JavaType;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * A {@link Deserializers} implementation that implements additional type checks
+ * to harden deserialisation against security issues.
+ *
+ * Also see {@link org.codehaus.jackson.map.module.DeserializationSecurityModule}.
+ *
+ */
+public class SecurityBeanDeserializer extends Deserializers.Base
+{
+
+    protected final static Set<String> BLOCKED_CLASS_NAMES;
+
+    static {
+        Set<String> s = new HashSet<String>();
+        s.add(Object.class.getName());
+        s.add(Comparable.class.getName());
+        s.add(Serializable.class.getName());
+        BLOCKED_CLASS_NAMES = s;
+    }
+
+    @Override
+    public JsonDeserializer<?> findBeanDeserializer(final JavaType type, final DeserializationConfig config,
+            final DeserializerProvider provider,
+            final BeanDescription beanDesc,
+            final BeanProperty property)
+            throws JsonMappingException
+    {
+        if (
```

Comment:
I think check here should be strictly limited to `type`, since key and value deserializers are constructed separately (key deserializers via `KeyDeserializers`; value deserializers using method appropriate for type, mostly `findBeanDeserializer`).

`property` will be `null` for root values as well, so it shouldn't be needed.